src/wp-includes/blocks.php

```diff
@@ -933,27 +963,53 @@ function traverse_and_serialize_block( $block, $callback ) {
 }
 
 /**
- * Traverses the blocks applying transformations using the callback provided,
- * and returns a joined string of the aggregate serialization of the given parsed blocks.
+ * Given an array of parsed block trees, applies callbacks before and after serializing them and
+ * returns their concatenated output.
+ *
+ * Recursively traverses the blocks and their inner blocks and applies the two callbacks provided as
+ * arguments, the first one before serializing a block, and the second one after serializing.
+ * If either callback returns a string value, it will be prepended and appended to the serialized
+ * block markup, respectively.
+ *
+ * The callbacks will receive a reference to the current block as their first argument, so that they
```

Comment:
In the code there is a note that no parent element gets passed for this function. I also added a question about the next and previous as they aren't currently passed to the callback.

Reply:
Good point!

Reply:
ff183b972a

Reply:
Passing `$prev` and `$next` per ff183b972aea34e862b65cfe5d15b9a818620e3b. 

As for the PHPDoc, I think I'll keep it as-is, since the callbacks _will_ get the parent arg at all levels of the tree except for the top level 🙂